openedx/features/course_experience/tests/views/test_course_home.py

```diff
@@ -109,3 +120,56 @@ def test_queries(self):
             with check_mongo_calls(4):
                 url = course_home_url(self.course)
                 self.client.get(url)
+
+
+@ddt.ddt
+class TestCourseHomePageAccess(CourseHomePageTestCase):
+    """
+    Test access to the course home page.
+    """
+    @override_waffle_flag(UNIFIED_COURSE_TAB_FLAG, active=True)
+    @override_waffle_flag(SHOW_REVIEWS_TOOL_FLAG, active=True)
+    @ddt.data(
+        'anonymous',
+        'unenrolled',
+        'enrolled',
+    )
+    def test_home_page(self, user_type):
+        is_enrolled = user_type is 'enrolled'
+
+        # Set up the test user
+        if user_type is not 'anonymous':
+            self.user = UserFactory(password=TEST_PASSWORD)
+            self.client.login(username=self.user.username, password=TEST_PASSWORD)
+            if is_enrolled:
+                CourseEnrollment.enroll(self.user, self.course.id)
+
+        # Make this a verified course so that an upgrade message might be shown
+        add_course_mode(self.course, upgrade_deadline_expired=False)
+
+        # Create a welcome message
+        create_course_update(self.course, self.user, TEST_WELCOME_MESSAGE)
+
+        # Render the course home page
+        url = course_home_url(self.course)
+        response = self.client.get(url)
+
+        # Verify that the course tools and dates are always shown
+        self.assertContains(response, 'Course Tools')
+        self.assertContains(response, 'Today is')
+
+        # Verify that the outline, start button, course sock, and welcome message
+        # are only shown to enrolled users.
+        expected_count = 1 if is_enrolled else 0
+        self.assertContains(response, TEST_CHAPTER_NAME, count=expected_count)
+        self.assertContains(response, 'Start Course', count=expected_count)
+        self.assertContains(response, 'Learn About Verified Certificate', count=expected_count)
+        self.assertContains(response, TEST_WELCOME_MESSAGE, count=expected_count)
+
+    def test_sign_in_button(self):
+        """
+        Verify that the sign in button will return to this page.
```

Comment:
I couldn't think of a more general place, because we don't have tests for templates. Every page is essentially equally valid. It seemed reasonable to me that this behavior was important for the course experience. I don't see the need for a comment.